scripts/migrations/039_projecttodo_to_obligationlist.py

```diff
@@ -0,0 +1,188 @@
+db = couch[DBNAME]
+userdb = couch[USER_DBNAME]
+
+# ----------------------------------------
+# queries
+# ----------------------------------------
+
+# get all projects with todos
+all_project_with_todos = {"selector": {"type": {"$eq": "project"}, "todos": {"$exists": True}}, "limit":20000}
+
+# ---------------------------------------
+# functions
+# ---------------------------------------
+
+def getOblTextNLevel(oblId):
+    obltextNLevel = {}
+    get_obligation_using_id = {"selector": {"type": {"$eq": "obligation"}, "_id": {"$eq": ""+oblId+"" }}}
+    obligation = db.find(get_obligation_using_id)
+    for obl in obligation:
+        text = obl.get("text")
+        obllevel = obl.get("obligationLevel")
+        obltextNLevel["oblText"] = text
+        obltextNLevel["oblLevel"] = obllevel
+    
+    return obltextNLevel
+
+def getDateOnly(dateWithTimes):
+    return dateWithTimes.split()[0]
+
+def getEmailFromUserId(userid):
+    emailId = ""
+    get_user_by_id = {"selector": {"type": {"$eq": "user"}, "_id": {"$eq": ""+userid+"" }}}
+    user = userdb.find(get_user_by_id)
+    for usr in user:
+        emailId = usr.get("email")
+
+    return emailId
+
+def getLinkedObligationList(linkedOblId):
+    oblList = []
+    get_obligationList_by_id = {"selector": {"type": {"$eq": "obligationList"}, "_id": {"$eq": ""+linkedOblId+"" }}}
+    obligationList = db.find(get_obligationList_by_id)
+    for obList in obligationList:
+        oblList = obList
+    
+    return oblList
+
+def mergeTodoWithObligationList(resultFile, all_projects):
+    log = {}
+    log['totalCount'] = len(all_projects)
+    log['updatedProjectsWithobligationList'] = []
+
+    oblList = {}
+    obligationStatusInfo = {}
+    obligationStatusInfos = {}
+
+    for project in all_projects:
+        linkedOblId = project.get("linkedObligationId")
+        todos = project.get("todos")
+        if len(todos) > 0:
+            if linkedOblId is None:
+                oblList["type"] = 'obligationList'
+                oblList["projectId"] = project.get("_id")
+                for todo in todos:
+                    todotext = getOblTextNLevel(todo.get("todoId")).get("oblText")
+                    obligationLevel = getOblTextNLevel(todo.get("todoId")).get("oblLevel")
+                    obligationStatusInfo["obligationLevel"] = obligationLevel
+                    obligationStatusInfo["text"] = todotext
+                    obligationStatusInfo["status"] = 'FULFILLED' if todo.get("fulfilled") == True else 'OPEN'
+                    todocomment = todo.get("comments")
+                    obligationStatusInfo["comment"] = todocomment if todocomment is not None else ""
+                    obligationStatusInfo["modifiedOn"] = getDateOnly(todo.get("updated")) if todo.get("updated") != "" else ""
+                    obligationStatusInfo["modifiedBy"] = getEmailFromUserId(todo.get("userId"))
+                    obligationStatusInfo["id"] = todo.get("todoId")
```

Comment:
(from discussion) - Id in obligationStatusInfo is for external id and not couch db doc id

Reply:
removed it